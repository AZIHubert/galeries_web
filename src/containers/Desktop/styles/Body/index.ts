import styled from 'styled-components';

const Body = styled.div`
  flex-grow: 1;
`;

export default Body;
