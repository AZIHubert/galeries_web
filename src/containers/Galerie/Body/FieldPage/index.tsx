import * as React from 'react';

const FieldPage = () => (
  <div>
    <button>
      add a picture
    </button>
  </div>
);

export default FieldPage;
