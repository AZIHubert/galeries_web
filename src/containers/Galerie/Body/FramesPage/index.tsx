import * as React from 'react';

import Modal from '#components/Modal';

import { GalerieContext } from '#contexts/galerieContext';

import FrameModal from './FrameModal';

import {
  Container,
} from './styles';

import AddFrameButton from './AddFrameButton';
import Frame from './Frame';

const FramesPage = () => {
  const { galerie } = React.useContext(GalerieContext);
  const [open, setOpen] = React.useState<boolean>(false);

  const handleClose = () => setOpen(false);
  const handleOpen = () => {
    if (galerie) {
      setOpen(true);
    }
  };

  const frames = galerie ? galerie.frames.frames : {};

  return (
    <Container>
      <div>
        {!!galerie && Object.keys(frames).sort(
          (a, b) => (
            new Date(frames[b].createdAt).getTime()
        - new Date(frames[a].createdAt).getTime()
          ),
        ).map((index) => (
          <Frame
            key={frames[index].id}
            frame={frames[index]}
          />
        ))}
      </div>
      <AddFrameButton
        onClick={handleOpen}
      />
      <Modal.Portal
        handleClose={handleClose}
        open={open}
      >
        <FrameModal
          handleClose={handleClose}
        />
      </Modal.Portal>
    </Container>
  );
};

export default FramesPage;
