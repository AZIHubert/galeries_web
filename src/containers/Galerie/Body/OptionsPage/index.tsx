import * as React from 'react';

const OptionsPage = () => (
  <div>
    option
  </div>
);

export default OptionsPage;
