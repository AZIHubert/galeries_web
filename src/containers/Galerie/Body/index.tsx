import * as React from 'react';

import {
  ButtonMenu,
  Container,
  Menu,
} from './styles';

interface BodyI {
  galerie: GalerieI;
}

const Body = ({
  galerie,
}: BodyI) => {
  const [page, setPage] = React.useState<'field' | 'users' | 'options'>('field');

  return (
    <Container>
      <Menu>
        <ButtonMenu
          current={page === 'field'}
          onClick={() => setPage('field')}
          margin
        >
            Field
        </ButtonMenu>
        <ButtonMenu
          current={page === 'users'}
          onClick={() => setPage('users')}
          margin
        >
              Users
        </ButtonMenu>
        {galerie.role !== 'user' ? (
          <ButtonMenu
            current={page === 'options'}
            onClick={() => setPage('options')}
          >
                Options
          </ButtonMenu>
        ) : null}
      </Menu>
    </Container>
  );
};

export default Body;
