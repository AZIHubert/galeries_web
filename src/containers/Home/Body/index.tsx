import * as React from 'react';

import { LogoGaleries } from '#ressources/svgComponents';

const Body = () => (
  <div>
    <h2>
        Welcome to
    </h2>
    <h1>
      <LogoGaleries />
    </h1>
    <p>
      A web app to share
      pictures with
      your friends and family.
    </p>
  </div>
);

export default Body;
