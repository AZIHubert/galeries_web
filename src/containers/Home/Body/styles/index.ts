export { default as Container } from './Container';
export { default as GaleriesLogo } from './GaleriesLogo';
