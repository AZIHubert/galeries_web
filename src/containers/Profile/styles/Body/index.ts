import styled from 'styled-components';

import mediaQueries from '#helpers/mediaQueries';

const Body = styled.div`
  flex-grow: 1;
  padding-top: ${({ theme }) => (`${theme.header.dashboard.height.small + 70}px`)};
  @media ${mediaQueries.mobileL} {
    padding-top: ${({ theme }) => (`${theme.header.dashboard.height.small + 80}px`)};
  }
  @media ${mediaQueries.laptop} {
    padding-top: ${({ theme }) => (`${theme.header.dashboard.height.small + 90}px`)};
  }
  @media ${mediaQueries.laptopL} { 
    padding-top: ${({ theme }) => (`${theme.header.dashboard.height.small + 120}px`)};
  }
`;

export default Body;
