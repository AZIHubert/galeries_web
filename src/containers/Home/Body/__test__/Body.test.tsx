import { cleanup } from '@testing-library/react';
import * as React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';

import ThemeProvider from '#contexts/ThemeContext';
import reducers from '#store/reducers';

import Body from '../index';

const mockedStore = createStore(reducers);

const Container = () => (
  <Provider store={mockedStore}>
    <ThemeProvider>
      <Body />
    </ThemeProvider>
  </Provider>
);

describe('Body', () => {
  afterEach(cleanup);
  it('renders without crashing', () => {
    const tree = renderer.create(<Container />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
