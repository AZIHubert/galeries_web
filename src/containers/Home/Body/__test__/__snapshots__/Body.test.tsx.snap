// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body renders without crashing 1`] = `
<div
  className="sc-jrAGrp hjYpvF"
>
  <div>
    <p
      className="sc-bdfBwQ jBbmrr"
      color="black"
      fontStyle="normal"
      fontWeight="bold"
    >
      Welcome to
    </p>
    <h1
      className="sc-kEjbxe dVcOQk"
    >
      <svg
        viewBox="0 0 356.12 51.77"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <defs>
          <linearGradient
            gradientUnits="userSpaceOnUse"
            id="logoGaleries_svg__linear-gradient"
            x1={0.96}
            x2={370.28}
            y1={25.85}
            y2={25.85}
          >
            <stop
              offset={0}
              stopColor="#23f2ff"
            />
            <stop
              offset={0.15}
              stopColor="#2ae9ff"
            />
            <stop
              offset={0.39}
              stopColor="#3ccfff"
            />
            <stop
              offset={0.7}
              stopColor="#5aa6ff"
            />
            <stop
              offset={0.99}
              stopColor="#7b78ff"
            />
          </linearGradient>
          <linearGradient
            id="logoGaleries_svg__linear-gradient-2"
            x1={0.96}
            x2={370.28}
            xlinkHref="#logoGaleries_svg__linear-gradient"
            y1={25.99}
            y2={25.99}
          />
          <linearGradient
            id="logoGaleries_svg__linear-gradient-3"
            x1={0.96}
            x2={370.28}
            xlinkHref="#logoGaleries_svg__linear-gradient"
            y1={26.02}
            y2={26.02}
          />
          <linearGradient
            id="logoGaleries_svg__linear-gradient-5"
            x1={0.96}
            x2={370.28}
            xlinkHref="#logoGaleries_svg__linear-gradient"
            y1={25.98}
            y2={25.98}
          />
          <linearGradient
            id="logoGaleries_svg__linear-gradient-6"
            x1={0.96}
            x2={370.28}
            xlinkHref="#logoGaleries_svg__linear-gradient"
            y1={25.92}
            y2={25.92}
          />
          <linearGradient
            id="logoGaleries_svg__linear-gradient-8"
            x1={0.96}
            x2={370.28}
            xlinkHref="#logoGaleries_svg__linear-gradient"
            y1={26.16}
            y2={26.16}
          />
          <style>
            .logoGaleries_svg__cls-2{fill:url(#logoGaleries_svg__linear-gradient-2)}
          </style>
        </defs>
        <g
          data-name="Layer 2"
          id="logoGaleries_svg__Layer_2"
        >
          <g
            data-name="Layer 1"
            id="logoGaleries_svg__Layer_1-2"
          >
            <path
              d="M35.15 16.49a9.39 9.39 0 00-1.36-5c-2-3.27-5.93-3.81-8.24-3.81-12 0-14.38 11.17-14.38 18.73 0 3.34.48 9.33 4.09 13.22 2.45 2.72 5.72 4.22 11.58 4.22 5.11 0 6.88-.61 8-.95V32.56h-9.5v-8.17h20.1V49c-3.07 1-8.11 2.72-18 2.72-11.24 0-17-2.72-20.91-6.54C1 39.78 0 32 0 26.77 0 9.33 10.9 0 25.82 0c6.33 0 12.87 1.77 16.76 6.2 3.27 3.75 3.47 7.63 3.54 10.29z"
              fill="url(#logoGaleries_svg__linear-gradient)"
            />
            <path
              className="logoGaleries_svg__cls-2"
              d="M85.15 1.36l17.1 49.25H91.62L88.08 40H68.59l-3.67 10.61H54.83L72.61 1.36zM85.42 32l-6.81-21-7.29 21z"
            />
            <path
              d="M112.46 1.43h10.76v40.81h21.12v8.37h-31.88z"
              fill="url(#logoGaleries_svg__linear-gradient-3)"
            />
            <path
              className="logoGaleries_svg__cls-2"
              d="M154.76 1.36h33v8h-22.44v11.76h21v8h-21v13.32h23.16v8.17h-33.72z"
            />
            <path
              d="M199.52 1.36h18.87c5.45-.07 11-.07 14.91 3.34a12.48 12.48 0 014.3 9.47c0 9.2-7.91 11.38-10.29 12.06 7.42.68 8 4.9 8.86 11.31a94.56 94.56 0 001.49 9.53 8.92 8.92 0 001.5 3.54h-11.65c-.47-2.72-2-15.19-2.65-17.09-1-2.8-3.55-2.8-5.32-2.8h-9.26v19.89h-10.76zm10.42 21.53h8.51c1.23 0 3.68 0 5.39-1.36a6.85 6.85 0 002.72-5.73c0-3.33-1.63-4.9-2.59-5.51-1.63-1.16-3.61-1.16-6.95-1.16h-7.08z"
              fill="url(#logoGaleries_svg__linear-gradient-5)"
            />
            <path
              d="M249 1.23h10.6v49.38H249z"
              fill="url(#logoGaleries_svg__linear-gradient-6)"
            />
            <path
              className="logoGaleries_svg__cls-2"
              d="M273 1.36h33v8h-22.43v11.76h21v8h-21v13.32h23.16v8.17H273z"
            />
            <path
              d="M326.83 34.88c0 2.32-.14 9.33 8.92 9.33 5.72 0 9.33-2.72 9.33-6.88 0-4.36-3.06-5.11-9.33-6.61-11.72-2.93-18.12-6.4-18.12-14.92 0-8.1 5.79-15.25 19.48-15.25 5.11 0 11.17 1.15 14.85 5.79a14.15 14.15 0 012.8 9.12h-10.15c-.14-1.7-.69-7.49-8-7.49-4.91 0-8.11 2.66-8.11 6.68 0 4.56 3.61 5.38 10.69 7.15 8.52 2 16.9 4.36 16.9 14 0 8.86-7.36 16-21 16-18.87 0-19-12.12-19.08-16.89z"
              fill="url(#logoGaleries_svg__linear-gradient-8)"
            />
          </g>
        </g>
      </svg>
    </h1>
    <p
      className="sc-bdfBwQ iHpmfc"
      color="black"
      fontStyle="italic"
      fontWeight="normal"
    >
      A web app to share
    </p>
    <p
      className="sc-bdfBwQ iHpmfc"
      color="black"
      fontStyle="italic"
      fontWeight="normal"
    >
      pictures with
    </p>
    <p
      className="sc-bdfBwQ iHpmfc"
      color="black"
      fontStyle="italic"
      fontWeight="normal"
    >
      your friends and family.
    </p>
  </div>
</div>
`;
